idea_town/frontend/static-src/app/main.js

```diff
@@ -18,32 +20,50 @@ app.extend({
     fetch('/api/me?format=json', {
       credentials: 'same-origin'
     }).then((response) => response.json()).then((userData) => {
+      app.on('webChannel:addon-available', () => {
+        app.me.hasAddon = true;
+        app.experiments = new ExperimentsCollection();
+        app.experiments.fetch({
+          success: app.blastOff
+        });
+      });
       app.me = new Me({
         user: userData
       });
+    }).catch((err) => {
+      // for now, log the error in the console & do nothing in the UI
+      console && console.error(err); // eslint-disable-line no-console
+    });
+  },
 
-      app.experiments = new ExperimentsCollection();
-      app.experiments.fetch();
-
-      // session won't change without a hard refresh, but addon state could, so:
-      // if addon state changes, dump user back to '/' and let the router handle
-      // redirecting to the correct landing page
-      app.me.on('change:hasAddon', () => { app.router.reload(); });
+  blastOff() {
+    app.pageManager = new PageManager({
+      pageContainer: document.querySelector('[data-hook=page-container]')
+    });
 
-      app.pageManager = new PageManager({
-        pageContainer: document.querySelector('[data-hook=page-container]')
-      });
+    app.router = new Router();
 
-      app.router = new Router();
+    if (!app.router.history.started()) {
       app.router.history.start();
-    }).catch((err) => {
-      // for now, log the error in the console & do nothing in the UI
-      console && console.error(err); // eslint-disable-line no-console
+    }
+
+    app.me.on('change:hasAddon', () => {
+      app.router.reload();
     });
+  },
+
+  // This is how you navigate around the app.
+  // this gets called by a global click handler that handles
+  // all the <a> tags in the app.
+  // it expects a url without a leading slash.
```

Comment:
Would you mind explaining this a bit as well? It seems like all you're doing is stripping leading slashes and calling `router.history.navigate`. Could we configure the router to automatically strip leading slashes, or integrate in a more pluggable way (maybe a middleware-ish approach)?